### PR TITLE
code health rc4test

### PR DESCRIPTION
**Pull request by pauli-oracle**
- [x] tests are added or updated

Update rc4test to use the test infrastructure

### CODE REVIEW

File: test/rc4test.c [hunk 1]
```diff
@@ -61,115 +54,74 @@ static unsigned char output[7][30] = {
...
-                printf(" %02x", expected[j]);
-            printf("\n");
-            err++;
-        } else
-            printf("ok\n");
+    RC4_set_key(&key, keys[3][0], &(keys[3][1]));
+    memset(obuf, 0, sizeof(obuf));
+    RC4(&key, i, &(data[3][0]), obuf);
+    RC4(&key, data_len[3] - i, &(data[3][i]), &(obuf[i]));
+    return TEST_mem_eq(obuf, data_len[3] + 1, output[3], data_len[3] + 1);
```
Comment (mattcaswell) [inline test/rc4test.c]: The original test also had a check for `obuf[i] != 0` - presumably to check for overruns.

File: test/rc4test.c [hunk 2]
```diff
@@ -61,115 +54,74 @@ static unsigned char output[7][30] = {
...
+    memset(buf, 0, sizeof(buf));
+    SHA1_Init(&c);
+    for (i = 0; i < 2571; i++) {
+        RC4(&key, sizeof(buf), buf, buf);
+        SHA1_Update(&c, buf, sizeof(buf));
     }
-    EXIT(err);
+    SHA1_Final(md, &c);
+
+    return TEST_mem_eq(md, sizeof(md), expected, sizeof(md));
```
Comment (mattcaswell) [inline test/rc4test.c]: probably really should be `TEST_mem_eq(md, sizeof(md), expected, sizeof(expected));`

File: test/rc4test.c (hunk 1 above)
Comment (pauli-oracle) [inline test/rc4test.c]: Added back in, thanks.

**Comment by richsalz** [comment]
+1

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**